[DO NOT MERGE] Eager workflow start public preview docs (#3689)
temporalio/documentation: docs/develop/worker-performance.mdx

  - Performance
---

import { CaptionedImage } from '@site/src/components';

> import * as Components from '@site/src/components';

Author: fixed

These Workflows have a happy path that needs to initiate interactions within low tens of milliseconds, but they also want to take advantage of server-driven retries, and reliable compensation processes, for those less happy days.

**Quick Start**

Eager Workflow Start requires the Starter and the Worker to share a Client located in the same process and setting the `request_eager_start` (or similar name) to true in the Start Workflow call.
When set, and the Worker has a Workflow Task slot available and the Workflow Definition registered, the Worker can execute the first task of the Workflow locally without first making a round-trip to the Temporal Server.
This is typically most useful in combination with a Local Activity executing in the first Workflow Task, since other Workflow API calls that require waiting on something will force a round-trip.

:::tip RESOURCES

- [Go SDK - Code sample](https://github.com/temporalio/samples-go/tree/main/eager-workflow-start)
- [Java SDK - Code sample](https://github.com/temporalio/samples-java/blob/main/core/src/main/java/io/temporal/samples/hello/HelloEagerWorkflowStart.java)

> - Python SDK - use something like `request_eager_start` in your Workflow-Start-Options

Author: thanks 👍

> - .NET SDK - use something like `request_eager_start` in your Workflow-Start-Options

Reviewer: ```suggestion
- .NET SDK - use `RequestEagerStart` in your `WorkflowOptions` when starting a workflow
```

> - .NET SDK - use something like `request_eager_start` in your Workflow-Start-Options

Reviewer 2: RequestEagerStart
